distributed/deploy/ssh.py

```diff
@@ -256,8 +264,9 @@ def __init__(self, scheduler_addr, scheduler_port, worker_addrs, nthreads = 0, n
 
         # Start worker nodes
         self.workers = []
-        for i, addr in enumerate(worker_addrs):
-            self.add_worker(addr)
+        for addr, nthreads, nprocs in zip(worker_addrs, self.threads,
+                                          self.nprocs):
+            self.add_worker(addr, nthreads, nprocs)
```

Comment:
Yes, that it what I mean.  Or, if not all features, it should at least be pretty trivial to extend this to almost all features, ideally without a long chain of if-else statements.  I hesitate to add any user-facing feature where some things work and some very related things don't work.  This tends to fill my inbox with confusing user reports :)